### PR TITLE
Added migration for the resources db.

### PR DESCRIPTION
**Pull request by aditya-07**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1902 

**Description**
1. Updated version of db as part of the changed for  #1902 .
2. Added migration for the db.
3. Updated code to export schema (required for tests).
4. Added tests.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?
No

**Type**
Choose one: Bug fix

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by aditya-07** [comment]
@joiskash FYI

File: engine/src/main/java/com/google/android/fhir/db/impl/ResourceDatabase.kt [hunk 1]
```diff
@@ -48,11 +50,18 @@ import com.google.android.fhir.db.impl.entities.UriIndexEntity
...
+  exportSchema = true
 )
 @TypeConverters(DbTypeConverters::class)
 internal abstract class ResourceDatabase : RoomDatabase() {
   abstract fun resourceDao(): ResourceDao
   abstract fun localChangeDao(): LocalChangeDao
 }
+
+val MIGRATION_1_2 =
+  object : Migration(1, 2) {
```
Comment (santosh-pingle) [inline engine/src/main/java/com/google/android/fhir/db/impl/ResourceDatabase.kt:63]: optional:
```suggestion
 object : Migration(/* startVersion = */ 1, /* endVersion = */ 2) {
```

File: engine/src/androidTest/java/com/google/android/fhir/db/impl/ResourceDatabaseMigrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+@RunWith(AndroidJUnit4::class)
+class ResourceDatabaseMigrationTest {
+
+  @get:Rule
+  val helper: MigrationTestHelper =
+    MigrationTestHelper(InstrumentationRegistry.getInstrumentation(), ResourceDatabase::class.java)
+
+  @Test
+  @Throws(IOException::class)
+  fun migrate1To2_should_not_throw_exception() {
```
Comment (jingtang10) [inline engine/src/androidTest/java/com/google/android/fhir/db/impl/ResourceDatabaseMigrationTest.kt:37]: can you save a patient to db v1 and load the patient from db v2 after migration and verify that they're the same.

File: engine/build.gradle.kts [hunk 3]
```diff
@@ -126,6 +135,7 @@ dependencies {
   testImplementation(Dependencies.mockitoKotlin)
   testImplementation(Dependencies.mockWebServer)
   testImplementation(Dependencies.robolectric)
+  androidTestImplementation(Dependencies.Room.testing)
```
Comment (jingtang10) [inline engine/build.gradle.kts:138]: put this in the right block

File: engine/src/androidTest/java/com/google/android/fhir/db/impl/ResourceDatabaseMigrationTest.kt (hunk 2 above)
Comment (yigit) [reply engine/src/androidTest/java/com/google/android/fhir/db/impl/ResourceDatabaseMigrationTest.kt:37]: yes, these are the kind of tests we need to write to ensure data is migrated properly.

**Review by jingtang10 — APPROVED** [review]

File: engine/build.gradle.kts (hunk 3 above)
Comment (aditya-07) [reply engine/build.gradle.kts]: Done

**Pull request merged**